src/praisonai/tests/unit/test_sequential_tool_calling.py

```
@@ -0,0 +1,679 @@
+#!/usr/bin/env python3
+"""
+Comprehensive mock tests for sequential tool calling functionality in PraisonAI.
+This test suite covers various scenarios of sequential tool execution.
+"""
+
+import sys
+import os
+import pytest
+import json
+from unittest.mock import patch, Mock, MagicMock
```

Ruff (0.11.9)

`unittest.mock.Mock` imported but unused

Remove unused import

(F401)